code/script.js

```diff
@@ -26,22 +34,130 @@ const showMessage = (message, sender) => {
       </section>
     `
   }
+
   // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
   chat.scrollTop = chat.scrollHeight
 }
 
 // Starts here
 const greeting = () => {
-  showMessage(`Hello there, What's your name?`, 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆
+  showMessage(`Welcome to the Travelbot! What's your name?`, 'bot')
+}
+
+const continentQuestion = () => {
+  showMessage(`Hello ${yourName}! Which continent would you like to visit?`, 'bot')
+  inputWrapper.innerHTML = /*html*/`
+    <select id="select">
+      <option disabled selected value="">Pick a continent</option>
+      <option id="africa" value="Africa">Africa</option>
+      <option id="europe" value="Europe">Europe</option>
+      <option id="south-america" value="South America">South America</option>
+      <option id="north-america" value="North America">North America</option>
+      <option id="asia" value="Asia">Asia</option>
+      <option id="antarctica" value="Antarctica">Antarctica</option>
+      <option id="oceania" value="Oceania">Oceania</option>
+   </select>
+  `
+  const select = document.getElementById('select')
+  select.addEventListener('change', () => handleContinentQuestion(select.value))
+}
+
+const weatherQuestion = () => {
+  showMessage('What kind of weather do you prefer?', 'bot')
+  inputWrapper.innerHTML = /*html*/ `
+  <button id="sun" value="sun">Sunny</button>
+  <button id="cloud" value="cloud">Cloudy</button>
+  <button id="snow" value="snow">Snowy</button>
+  `
+  document
+  .getElementById('sun')
+  .addEventListener('click', () => handleWeatherQuestion('sunny'))
+  document
+  .getElementById('cloud')
+  .addEventListener('click', () => handleWeatherQuestion('cloudy'))
+  document
+  .getElementById('snow')
+  .addEventListener('click', () => handleWeatherQuestion('snowy'))
+}
+
+
+const bookQuestion = () => {
+  showMessage(`Perfect! We have found a destination in ${continent} where the weather is always as you want. Would you like to book a ticket?`, 'bot')
+  inputWrapper.innerHTML = /*html*/ `
+  <button id="yes" value="yes">Hell yeah!</button>
+  <button id="no" value="no">No, thanks.</button>
+  `
+  document
+  .getElementById('yes')
+  .addEventListener('click', () => handleBookQuestion('Hell yeah!'))
+  document
+  .getElementById('no')
+  .addEventListener('click', () => handleBookQuestion('No, thanks.'))
+}
+
+const lastMessage = () => {
+    showMessage(`Thank you, bye!`, 'bot')
+    inputWrapper.innerHTML = ``
+}
+
+// const lastMessage = () => {
+//   if (book === 'yes'){
+//     showMessage(`Thank you ${yourName}! Your booking has been sent tou you`, 'bot')
+//     inputWrapper.innerHTML = ''
+//   } else {
+//     showMessage(`Sorry to see you go, ${yourName}`, 'bot')
+//     inputWrapper.innerHTML = ''
+//   }
+// }
```

Comment:
It's really nice I think that you've left this commented out rather than just deleting it, so if you come back to this at the end of the course you'll be able to tell past you exactly how to solve it! I think there's a couple of different ways to get it working. Since you've already been using global variables for other things, you could also create a global 'book' variable which you assign to 'yes' at line 92 under the 'yes' button event listener. Or, since you already passed down the text 'Hell yeah!' into the handleBookQuestion function, you could continue to pass it through to lastMessage when you call it on 155, and then use the following code instead:
```
const lastMessage = (book) => {
  if (book === 'Hell yeah!'){
     showMessage(`Thank you ${yourName}! Your booking has been sent to you`, 'bot')
     inputWrapper.innerHTML = ''
   } else if (book ===  'No, thanks.' {
     showMessage(`Sorry to see you go, ${yourName}`, 'bot')
     inputWrapper.innerHTML = ''
   }
 }
```